cgen error for Color.from_string()

<!-- It is advisable to update all relevant modules using `v outdated` and `v install` -->
**V doctor:**
```
V full version: V 0.4.0 2fec2fa.c5e1c8e
OS: linux, Ubuntu 20.04.6 LTS
Processor: 4 cpus, 64bit, little endian, Intel(R) Core(TM) i3-3225 CPU @ 3.30GHz

getwd: /v/vnew
vexe: /v/vnew/v
vexe mtime: 2023-08-17 13:54:55

vroot: OK, value: /v/vnew
VMODULES: OK, value: /home/delian/.vmodules
VTMP: OK, value: /tmp/v_1000

Git version: git version 2.41.0
Git vroot status: weekly.2023.33-5-gc5e1c8ee
.git/config present: true

CC version: cc (Ubuntu 10.5.0-1ubuntu1~20.04) 10.5.0
thirdparty/tcc status: thirdparty-linux-amd64 12f392c3

```

**What did you do?**
`v -g -o vdbg cmd/v && vdbg zz.v`
```v
enum Color {
        red
        green
        blue
}
Color.from_string()

```

**What did you expect to see?**

a checker error, for the invalid usage of Color.from_string() (the generated function expects 1 parameter)

**What did you see instead?**
```
==================
/tmp/v_1000/zz.7612909021817861809.tmp.c:12480: error: too few arguments to function
...
==================
(Use `v -cg` to print the entire error message)

builder error: 
==================
C error. This should never happen.

This is a compiler bug, please report it using `v bug file.v`.

https://github.com/vlang/v/issues/new/choose

You can also use #help on Discord: https://discord.gg/vlang

```

--- a/vlib/v/checker/fn.v
+++ b/vlib/v/checker/fn.v
@@ -865,6 +865,15 @@ fn (mut c Checker) fn_call(mut node ast.CallExpr, mut continue_check &bool) ast.
 			}
 			idx := c.table.type_idxs[full_enum_name]
 			ret_typ := ast.Type(idx).set_flag(.option)
+			if node.args.len != 1 {
+				c.error('expected 1 argument, but got ${node.args.len}', node.pos)
+			} else {
+				node.args[0].typ = c.expr(mut node.args[0].expr)
+				if node.args[0].typ != ast.string_type {
+					styp := c.table.type_to_str(node.args[0].typ)
+					c.error('expected `string` argument, but got `${styp}`', node.pos)
+				}
+			}
 			node.return_type = ret_typ
 			return ret_typ
 		}

--- a/vlib/v/checker/tests/enum_from_string_args_err.out
+++ b/vlib/v/checker/tests/enum_from_string_args_err.out
@@ -1,0 +1,34 @@
+vlib/v/checker/tests/enum_from_string_args_err.vv:8:2: error: expected 1 argument, but got 0
+    6 |
+    7 | fn main() {
+    8 |     Color.from_string()
+      |     ~~~~~~~~~~~~~~~~~~~
+    9 |     Color.from_string('red', 'blue')
+   10 |     Color.from_string(22)
+vlib/v/checker/tests/enum_from_string_args_err.vv:9:2: error: expected 1 argument, but got 2
+    7 | fn main() {
+    8 |     Color.from_string()
+    9 |     Color.from_string('red', 'blue')
+      |     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+   10 |     Color.from_string(22)
+   11 |     Color.from_string(22.22)
+vlib/v/checker/tests/enum_from_string_args_err.vv:10:2: error: expected `string` argument, but got `int literal`
+    8 |     Color.from_string()
+    9 |     Color.from_string('red', 'blue')
+   10 |     Color.from_string(22)
+      |     ~~~~~~~~~~~~~~~~~~~~~
+   11 |     Color.from_string(22.22)
+   12 |     Color.from_string(true)
+vlib/v/checker/tests/enum_from_string_args_err.vv:11:2: error: expected `string` argument, but got `float literal`
+    9 |     Color.from_string('red', 'blue')
+   10 |     Color.from_string(22)
+   11 |     Color.from_string(22.22)
+      |     ~~~~~~~~~~~~~~~~~~~~~~~~
+   12 |     Color.from_string(true)
+   13 | }
+vlib/v/checker/tests/enum_from_string_args_err.vv:12:2: error: expected `string` argument, but got `bool`
+   10 |     Color.from_string(22)
+   11 |     Color.from_string(22.22)
+   12 |     Color.from_string(true)
+      |     ~~~~~~~~~~~~~~~~~~~~~~~
+   13 | }

--- a/vlib/v/checker/tests/enum_from_string_args_err.vv
+++ b/vlib/v/checker/tests/enum_from_string_args_err.vv
@@ -1,0 +1,13 @@
+enum Color {
+	red
+	green
+	blue
+}
+
+fn main() {
+	Color.from_string()
+	Color.from_string('red', 'blue')
+	Color.from_string(22)
+	Color.from_string(22.22)
+	Color.from_string(true)
+}

